src/radar_control.py

```diff
@@ -1016,29 +1061,66 @@ def main():
                 while time_remains:
                     # CFS block
                     if ave_params.num_sequences == 0 and aveperiod.cfs_flag:
-                        ave_params.sequence = aveperiod.cfs_sequence
-                        ave_params.decimation_scheme = (
-                            aveperiod.cfs_sequence.decimation_scheme
-                        )
-                        freq_spectrum = run_cfs_scan(
-                            radctrl_inproc_socket,
-                            radctrl_brian_socket,
-                            ave_params,
-                            driver_comms_socket,
-                        )
+                        if (
+                            cfs_params_dict[aveperiod].last_cfs_set_time
+                            < time.time() - ave_params.aveperiod.cfs_stable_time
+                        ):
+                            # Only let CFS run after the user set stable time has
+                            # passed to prevent CFS from switching freqs to quickly
+                            ave_params.sequence = aveperiod.cfs_sequence
+                            ave_params.decimation_scheme = (
+                                aveperiod.cfs_sequence.decimation_scheme
+                            )
 
-                        ave_params.cfs_masks = aveperiod.update_cfs_freqs(freq_spectrum)
-                        ave_params.aveperiod = aveperiod
-                        ave_params.num_sequences += 1
-                        ave_params.cfs_scan_flag = False
-                        ave_params.cfs_freq = freq_spectrum.cfs_freq
-                        ave_params.cfs_mags = [
-                            mag.cfs_data for mag in freq_spectrum.output_datasets
-                        ]
-                        for ind in range(len(aveperiod.cfs_slice_ids)):
-                            ave_params.cfs_range[aveperiod.cfs_slice_ids[ind]] = (
-                                aveperiod.cfs_range[ind]
+                            freq_spectrum = run_cfs_scan(
+                                radctrl_inproc_socket,
+                                radctrl_brian_socket,
+                                ave_params,
+                                driver_comms_socket,
                             )
+                            ave_params.num_sequences += 1
+                            ave_params.cfs_scan_flag = False
+
+                            if (
+                                not cfs_params_dict[aveperiod].set_new_freq
+                                and aveperiod.cfs_pwr_threshold
```

Comment:
Personally, I prefer `aveperiod.cfs_pwr_threshold is not None` as it is more clear that the field wasn't set